src/cares_wrap.cc

```diff
@@ -1064,30 +1064,28 @@ int ParseSoaReply(Environment* env,
   /* Can't use ares_parse_soa_reply() here which can only parse single record */
   unsigned int ancount = cares_get_16bit(buf + 6);
   unsigned char* ptr = buf + NS_HFIXEDSZ;
-  char* name;
-  char* rr_name;
+  MallocedBuffer<char> name;
```

Comment:
Since there is no further transfer of ownership, I think that is you use this pattern, i.e.:
```cpp
long temp_len;  // NOLINT(runtime/int)
char* name_tmp;
int status = ares_expand_name(ptr, buf, len, &name_tmp, &temp_len);
const ares_unique_ptr name(name_tmp);
```
You can use a `const ares_unique_ptr` getting more explicit code, and more compiler validation.
From [CppReference](https://en.cppreference.com/w/cpp/memory/unique_ptr):
> Only non-const unique_ptr can transfer the ownership of the managed object to another unique_ptr. If an object's lifetime is managed by a const std::unique_ptr, it is limited to the scope in which the pointer was created.